lib/Sema/SemaDeclCXX.cpp

```diff
@@ -6158,6 +6158,35 @@ static bool canPassInRegisters(Sema &S, CXXRecordDecl *D,
   return HasNonDeletedCopyOrMove;
 }
 
+static bool CheckTrivalDestructor(Sema &S, CXXRecordDecl *R,
```

Comment:
s/Trival/Trivial/
A bool-returning function named `Check...` strikes me as confusing. I think the meaning here is more like `bool HasTrivialDestructor`, except that `Has...` implies constness, whereas this function reserves the right to modify `*R` in the process of determining the triviality of its destructor.

Since `LookupSpecialMember` has the same semantics — "basically a getter, but might modify its argument anyway" — perhaps this function should be named similarly, something like `LookupTrivialityOfDestructor`.